### PR TITLE
Add Integratord options tag IT support

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| #3928            |

## Description

This Issue adds support for the new Slack integration `options` tag. This tag allows passing custom JSON string that will be sent as part of the message to the integration and/or replace values with the ones configured. The PR adds 20 test cases for different configurations and values.

### Added

- `test_integratord/test_integration_options/test_slack_options.py`
- `test_integratord/test_integration_options/test_shuffle_options.py`
- `test_integratord/test_integration_options/test_pagerduty_options.py`

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @Deblintrake09  (Developer)  | test_integration    | [🟢](https://ci.wazuh.info/job/Test_integration/41087/)[🟢](https://ci.wazuh.info/job/Test_integration/41088/) [🟢](https://ci.wazuh.info/job/Test_integration/41093/console) | [:green_circle::green_circle::green_circle:](https://github.com/wazuh/wazuh-qa/files/11478863/3928-results.zip)  |  Ubuntu Manager       | https://github.com/wazuh/wazuh-qa/pull/4166/commits/ed74782e1096d12f0c67b23153cd6d9eb49ee1f9        | Nothing to highlight |
| @damarisg  (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    expected_output:
+        - ".*(Enabling integration for: '{integration}')."
+        - ".*ERROR: Unable to run integration for ({integration}) -> integrations"
+        - '.*OS_IntegratorD.*(JSON file for options  doesn't exist)'
+        - '.*Response received.* [200].*'
+
+    '''
+    wazuh_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Read Response in ossec.log
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py:233]: I suggest you delete the comment, or if you add it, do not put the file name, but refer to the file in a general way.

File: tests/integration/test_integratord/test_integration_options/test_shuffle_options.py [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+    expected_output:
+        - ".*(Enabling integration for: '{integration}')."
+        - ".*ERROR: Unable to run integration for ({integration}) -> integrations"
+        - '.*OS_IntegratorD.*(JSON file for options  doesn't exist)'
+        - '.*Response received.* [200].*'
+
+    '''
+    wazuh_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Read Response in ossec.log
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_shuffle_options.py:233]: I suggest you delete the comment, or if you add it, do not put the file name, but refer to the file in a general way.

File: tests/integration/test_integratord/test_integration_options/test_slack_options.py [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    expected_output:
+        - ".*(Enabling integration for: '{integration}')."
+        - ".*ERROR: Unable to run integration for ({integration}) -> integrations"
+        - '.*OS_IntegratorD.*(JSON file for options  doesn't exist)'
+        - '.*Response received.* [200].*'
+
+    '''
+    wazuh_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Read Response in ossec.log
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_slack_options.py:233]: I suggest you delete the comment, or if you add it, do not put the file name, but refer to the file in a general way.

File: tests/integration/test_integratord/test_integration_options/test_slack_options.py [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+            brief: Truncate all the log files and json alerts files before and after the test execution.
+        - configure_local_internal_options_module:
+            type: fixture
+            brief: Configure the local internal options file.
+        - restart_wazuh_daemon_function:
+            type: fixture
+            brief: Restart wazuh daemon before starting a test.
+        - wait_for_start_module:
+            type: fixture
+            brief: Detect the start of the Integratord module in the ossec.log
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_slack_options.py:211]: If you add it, do not put the file name, but refer to the file in a general way.

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_integratord/test_integration_options/test_slack_options.py (hunk 4 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integration_options/test_slack_options.py]: Done in https://github.com/wazuh/wazuh-qa/pull/4166/commits/1460bad24bed27ff08df90a97d4af108d93e29ff

File: tests/integration/test_integratord/test_integration_options/test_slack_options.py (hunk 3 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integration_options/test_slack_options.py]: Done in https://github.com/wazuh/wazuh-qa/pull/4166/commits/1460bad24bed27ff08df90a97d4af108d93e29ff

File: tests/integration/test_integratord/test_integration_options/test_shuffle_options.py (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integration_options/test_shuffle_options.py]: Done in https://github.com/wazuh/wazuh-qa/pull/4166/commits/1460bad24bed27ff08df90a97d4af108d93e29ff

File: tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py (hunk 1 above)
Comment (Deblintrake09) [reply tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py]: Done in https://github.com/wazuh/wazuh-qa/pull/4166/commits/1460bad24bed27ff08df90a97d4af108d93e29ff

File: CHANGELOG.md [hunk 5]
```diff
@@ -18,6 +18,7 @@ Release report: TBD
 
 ### Added
 
+- Add callbacks and IT tests for Integratord options tag. ([#4108](https://github.com/wazuh/wazuh-qa/pull/4108)) \- (Framework + tests)
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply CHANGELOG.md]: Done in https://github.com/wazuh/wazuh-qa/pull/4166/commits/1460bad24bed27ff08df90a97d4af108d93e29ff

**Review by TomasTurina — APPROVED** [review]

File: tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t2_configurations, t2_configuration_metadata), ids=t2_case_ids)
+def test_pagerduty_options(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                           configure_local_internal_options_module, restart_wazuh_daemon_function,
+                           wait_for_start_module):
+    '''
+    description: Check that when configuring the options tag with differents values, the integration works as expected.
+                 The test also checks that when it is supposed to fail, it fails.
+
+    wazuh_min_version: 4.5.0
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py:174]: Could you change the wazuh_min_version to 4.6.0, please?

File: tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_pagerduty_no_option_tag(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                                 configure_local_internal_options_module, restart_wazuh_daemon_function,
+                                 wait_for_start_module):
+    '''
+    description: Check that when the options tag is not present for the PagerDuty integration, the integration works
+                 properly.
+
+    wazuh_min_version: 4.5.0
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_pagerduty_options.py:94]: Could you change the wazuh_min_version to 4.6.0, please?

File: tests/integration/test_integratord/test_integration_options/test_shuffle_options.py [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_integration_no_option_tag(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                                   configure_local_internal_options_module, restart_wazuh_daemon_function,
+                                   wait_for_start_module):
+    '''
+    description: Check that when the options tag is not present for the Shuffle integration, the integration works
+                 properly.
+
+    wazuh_min_version: 4.5.0
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_shuffle_options.py:94]: Could you change the wazuh_min_version to 4.6.0, please?

File: tests/integration/test_integratord/test_integration_options/test_shuffle_options.py [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t2_configurations, t2_configuration_metadata), ids=t2_case_ids)
+def test_shuffle_options(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                         configure_local_internal_options_module, restart_wazuh_daemon_function,
+                         wait_for_start_module):
+    '''
+    description: Check that when configuring the options tag with differents values, the Shuffle integration works as
+                 expected. The test also checks that when it is supposed to fail, it fails.
+
+    wazuh_min_version: 4.5.0
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_shuffle_options.py:174]: Could you change the wazuh_min_version to 4.6.0, please?

File: tests/integration/test_integratord/test_integration_options/test_slack_options.py [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t1_configurations, t1_configuration_metadata), ids=t1_case_ids)
+def test_integration_no_option_tag(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                                   configure_local_internal_options_module, restart_wazuh_daemon_function,
+                                   wait_for_start_module):
+    '''
+    description: Check that when the options tag is not present for the Slack integration, the integration works
+                 properly.
+
+    wazuh_min_version: 4.5.0
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_slack_options.py:94]: Could you change the wazuh_min_version to 4.6.0, please?

File: tests/integration/test_integratord/test_integration_options/test_slack_options.py [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata', zip(t2_configurations, t2_configuration_metadata), ids=t2_case_ids)
+def test_slack_options(configuration, metadata, set_wazuh_configuration, truncate_monitored_files,
+                       configure_local_internal_options_module, restart_wazuh_daemon_function,
+                       wait_for_start_module):
+    '''
+    description: Check that when configuring the options tag with differents values, the integration works as expected.
+                 The test also checks that when it is supposed to fail, it fails.
+
+    wazuh_min_version: 4.5.0
```
Comment (damarisg) [inline tests/integration/test_integratord/test_integration_options/test_slack_options.py:174]: Could you change the wazuh_min_version to 4.6.0, please?

**Review by damarisg — CHANGES_REQUESTED** [review]

**Review by TomasTurina — APPROVED** [review]

**Review by damarisg — APPROVED** [review]

File: CHANGELOG.md [hunk 12]
```diff
@@ -7,6 +7,10 @@ All notable changes to this project will be documented in this file.
 Wazuh commit: TBD \
 Release report: TBD
 
+### Added
+
+- Add callbacks and IT tests for Integratord options tag. ([#4108](https://github.com/wazuh/wazuh-qa/pull/4108)) \- (Framework + tests)
```
Comment (fcaffieri) [inline CHANGELOG.md:12]: This link is to a cancelled PR.

**Pull request merged**